src/tests/system.rs

```diff
@@ -607,3 +607,148 @@ mod fixed {
         }
     }
 }
+
+mod complex {
+    storage_types! {
+        types: Complex;
+
+        use crate::tests::*;
+
+        Q!(crate::tests, V);
+
+        #[test]
+        fn fp_categories() {
+            assert!(!Length::new::<meter>(
+                V::new(VV::infinity(), VV::infinity())
+            ).is_finite());
+            assert!(!Length::new::<meter>(
+                V::new(VV::neg_infinity(), VV::neg_infinity())
+            ).is_finite());
+            assert!(Length::new::<meter>(
+                V::new(VV::infinity(), VV::infinity())
+            ).is_infinite());
+            assert!(Length::new::<meter>(
+                V::new(VV::neg_infinity(), VV::neg_infinity())
+            ).is_infinite());
+            assert!(Length::new::<meter>(
+                V::new(VV::min_positive_value(), VV::min_positive_value())
+            ).is_normal());
+            assert!(Length::new::<meter>(
+                V::new(VV::max_value(), VV::max_value())
+            ).is_normal());
+            assert!(!Length::new::<meter>(
+                V::new(VV::zero(), VV::zero())
+            ).is_normal());
+            assert!(!Length::new::<meter>(
+                V::new(VV::nan(), VV::nan())
+            ).is_normal());
+            assert!(!Length::new::<meter>(
+                V::new(VV::infinity(), VV::infinity())
+            ).is_normal());
```

Comment:
This only partially makes sense for complex numbers, as you'd have to test eg. whether the complex number is_infinite() when _either_ the real or imaginary part is infinite. Should I leave it like this, expand to cover all cases or just remove it?